### PR TITLE
fix(build): use llvm/clang 12 to build the probe on Linux Kernel 5.x on Debian

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area build

> /area cmd

> /area pkg

> /area docs

> /area tests

**What this PR does / why we need it**:

Since Falco's eBPF probe supports clang 12 we can use it in the builder image. It appears that debian kernel 5.10 won't compile properly with clang 7, but works fine with 11+. See issue #106 for additional information.

PS: this PR contains the `buster` change from #110 , if this is accepted the other is superseded.

**Which issue(s) this PR fixes**:

https://github.com/falcosecurity/driverkit/issues/106

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by movd** [comment]
@leodido @fntlnz could you review this PR :) ?

File: build/builder.Dockerfile [hunk 1]
```diff
@@ -31,8 +31,18 @@ RUN apt-get update \
 	libudev-dev \
 	libpci-dev \
 	libiberty-dev \
+    lsb-release \
```
Comment (fntlnz) [inline build/builder.Dockerfile:34]: Please format accordingly

**Review by fntlnz — COMMENTED** [review]
Just a little thing to fix. Then we should be ok

File: build/builder.Dockerfile (hunk 1 above)
Comment (LucaGuerra) [reply build/builder.Dockerfile]: Done!

**Comment by LucaGuerra** [comment]
Thanks for the review @fntlnz ! Addressed the comment.

**Review by leodido — COMMENTED** [review]
/lgtm

**Comment by LucaGuerra** [comment]
Does Poiana need a bit more encouragement to merge? 🤔

**Comment by Issif** [comment]
@leodido @fntlnz can you try to unlock this merge please? We're stucked and you two are the only maintainers of this repo. Thanks

**Review by leodido — APPROVED** [review]

**Pull request merged**